timeout_handler.go

```diff
@@ -53,12 +63,14 @@ func (t *TimeoutHandler) GetTime() time.Time {
 	return t.now
 }
 
-func (t *TimeoutHandler) Tick(now time.Time) {
-	t.lock.Lock()
-	// update the time of the handler
-	t.now = now
-	t.lock.Unlock()
+func (t *TimeoutHandler) run() {
+	for t.shouldRun() {
+		<-t.ticks
```

Comment:
can we simultaneously select from ticks and close to immediately close and not wait for the tick?